src/util/conditionOperators/index.js

```diff
@@ -192,6 +192,15 @@ const filterComponentsForConditionComponentFieldOptions = (flattenedComponents)
         'button',
       ].includes(component.type);
 
+      // Change type for Day component with hidden fields to not treat it as a date
+      if (component.type === 'day' && (
+        component.fields.day.hide ||
+        component.fields.month.hide ||
+        component.fields.year.hide)
+      ) {
+        component.type = 'component';
+      }
+
```

Comment:
I'm concerned about this.  because it's a reference and you change the component type, it's effectively changed from `day` to `component` from this moment on.